### PR TITLE
Zeroconf discovery

### PR DESCRIPTION
**Pull request by DrTree**
Note:
~- PR is WIP and includes an unimpleented flag for "Exclude Time Clock"~

There's been a flurry of PRs so I'm just preparing this as a draft so I can get feedback if I've accidentally messed with any changes.

Ping @leaskovski @stikonas if you could review and comment I would be grateful.
Thanks in advance

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+__pycache__/
```
Comment (stikonas) [inline .gitignore:1]: End of file newline is missing

File: custom_components/heatmiserneo/config_flow.py [hunk 2]
```diff
@@ -37,31 +37,80 @@ class FlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
...
+        return self.async_show_form(
+            step_id="zeroconf_confirm",
+            description_placeholders={
+                'name': self._host
+            },
+        )
 
+    async def try_connection(self):
+        _LOGGER.debug("Trying connection...")
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
```
Comment (stikonas) [inline custom_components/heatmiserneo/config_flow.py:71]: Maybe `await asyncio.open_connection` is better here?

File: custom_components/heatmiserneo/translations/en.json [hunk 3]
```diff
@@ -4,14 +4,18 @@
...
             }
         },
-	"error": {
-	    "cannot_connect": "Failed to connect"
+        "error": {
+            "cannot_connect": "Failed to connect"
         }
     }
-}
+}
```
Comment (stikonas) [inline custom_components/heatmiserneo/translations/en.json:21]: newline

**Review by stikonas — COMMENTED** [review]
Looks alright, although haven't tested it yet.

File: custom_components/heatmiserneo/config_flow.py (hunk 2 above)
Comment (DrTree) [reply custom_components/heatmiserneo/config_flow.py:71]: Maybe but I say we leave it and then implement in a library with some kind of async hub.test()

Comment (stikonas) [reply custom_components/heatmiserneo/config_flow.py:71]: That's fine. At some point I can add it to https://git.stikonas.eu/andrius/neohub2 (alternatively merge requests are also welcome, at the moment I'm still adding other functionality)

**Comment by DrTree** [comment]
Ok, I've tried this for a while on my setup and it works just fine.
I've also removed and re-added using both workflows multiple times and tested on the devcontainer setup for HA using just the config flow (zero config and my docker setup are not friends 😢 )

**Comment by DrTree** [comment]
@MindrustUK I think this can be merged now

**Comment by MindrustUK** [comment]
Thanks for the PR.

**Pull request merged**